subsys/caf/modules/sensor_sampler.c

```diff
@@ -134,35 +153,34 @@ static bool is_active(const struct sensor_config *sc, struct sensor_data *sd,
 	return is_active;
 }
 
-static bool can_sensor_sleep(const struct sensor_config *sc,
-			     struct sensor_data *sd,
-			     const float *curr)
+static void process_sensor_activity(const struct sensor_config *sc,
+				    struct sensor_data *sd,
+				    const float *curr)
 {
 	size_t data_cnt = get_sensor_data_cnt(sc);
 	bool sleep = true;
 
 	for (size_t i = 0; i < data_cnt; i++) {
-		if (is_active(sc, sd, curr[i], sd->prev[i])) {
+		if (process_sensor_trigger_values(sc, sd, curr[i], sd->prev[i])) {
 			sleep = false;
 			break;
 		}
 	}
 
 	if (sleep) {
-		unsigned int timeout = sc->trigger->activation.timeout_ms;
-		unsigned int period = sc->sampling_period_ms;
-
-		sd->sleep_cnt++;
-		if (sd->sleep_cnt >= timeout/period) {
-			sd->sleep_cnt = 0;
-			return true;
+		if (sd->sleep_cntd > 0) {
+			--(sd->sleep_cntd);
+			LOG_DBG("  Sleep_cntd: %d", sd->sleep_cntd);
```

Comment:
Leading spaces are not needed

Reply:
Debugging leftover. It was there just to easily locate the right log message. Fixed.